### ISSUE TITLE
Expand Microsoft.Extensions.Hosting support to allow work to be done prior to Run*Async()

### ISSUE THREAD

**Issue by hellfirehd** [natemcmaster/CommandLineUtils#408]
**Is your feature request related to a problem? Please describe.**
I often have startup tasks that must be performed prior to running the console. I usually hang these off `IHost` so that they can create `IServiceScope` and benefit from the rest of the DI configuration.

```csharp
var host = CreateHostBuilder(args).Build();
await host.StartServiceBusAsync("QueueName");
await host.RunAsync().ConfigureAwait(false);
```

However, you can't call `.Buid()` twice so that doesn't work with `RunCommandLineApplicationAsync<T>()`.

**Describe the solution you'd like**
I want to be able to do this:

```csharp
class Program
{
    static async Task<Int32> Main(String[] args)
    {
        var host = CreateHostBuilder(args).Build();

        await host.StartServiceBus("QueueName");

        return await host.RunCommandLineApplicationAsync<Program>();
    }

    public static IHostBuilder CreateHostBuilder(String[] args) =>
        Host.CreateDefaultBuilder(args)
            .UseCommandLineApplication<Program>(args)
            .ConfigureServices((hostContext, services) =>
            {
                /* SNIP */
            });
}
```

**Describe alternatives you've considered**
I looked at #134 and #185. They both wanted the same thing I do but it was early days and it appears they kind of fizzled.

I have a pull request ready to go that implements the above. I'd love for you to take a look at it!

**Comment by natemcmaster** [comment]
The proposal seems fine to me. Let's continue to iterate on the implementation together as proposed in #409

**Comment by cbcrouse** [comment]
Hey @natemcmaster and @hellfirehd - I was able to address the issues of code duplication and missing tests. I've created [another PR](https://github.com/natemcmaster/CommandLineUtils/pull/415).

@hellfirehd your commits are still included in this, I just forked and cherry-picked.

**Comment by natemcmaster** [comment]
Resolved in #409, #415, and #416
